common/src/main/java/io/netty/util/collection/PrimitiveCollections.java

```diff
@@ -122,9 +142,12 @@ public boolean containsValue(Object value) {
      *
      * @param <V> the value type stored in the map.
      */
-    private static final class UnmodifiableIntObjectMap<V> implements IntObjectMap<V>,
-            Iterable<IntObjectMap.Entry<V>> {
+    private static final class UnmodifiableIntObjectMap<V> implements IntObjectMap<V> {
         final IntObjectMap<V> map;
+        Set<Integer> keySet;
```

Comment:
@Scottmitch I don't see why we should care about synthetic method generation. It's come up in a couple of code reviews and I would like to understand why we spend time discussing it?